src/LBFactory.sol

```diff
@@ -357,17 +352,17 @@ contract LBFactory is Ownable2Step, AccessControl, ILBFactory {
         }
 
         {
-            address implementation = _lbPairImplementation;
-
-            if (implementation == address(0)) revert LBFactory__ImplementationNotSet();
-
-            pair = ILBPair(
-                ImmutableClone.cloneDeterministic(
-                    implementation,
-                    abi.encodePacked(tokenX, tokenY, binStep),
-                    keccak256(abi.encode(tokenA, tokenB, binStep))
-                )
+            bytes memory data = abi.encodeWithSignature("initialize(uint16,uint16,uint16,uint16,uint24,uint16,uint24,uint24)",
+                preset.getBaseFactor(),
+                preset.getFilterPeriod(),
+                preset.getDecayPeriod(),
+                preset.getReductionFactor(),
+                preset.getVariableFeeControl(),
+                preset.getProtocolShare(),
+                preset.getMaxVolatilityAccumulator(),
+                activeId
             );
+            pair = ILBPair(address(new LBPairBeaconProxy(_lbPairBeacon, address(tokenX), address(tokenY), binStep, data)));
```

Comment:
Yes, the creation of the same pair (sorted) will not be possible, since there is a validation in the above
https://github.com/DistributedCollective/sovryn-lb-dex/blob/ab08ddafa24bc76b0d41843f366ac1d6fcf439ff/src/LBFactory.sol#L350-L352

This already covered by the unit tests as well
https://github.com/DistributedCollective/sovryn-lb-dex/blob/ab08ddafa24bc76b0d41843f366ac1d6fcf439ff/test/LBFactory.t.sol#L238

Reply:
discussed with cowsant in private, agreed on creation LBPairBeaconProxy conract deterministic + add reentrancy protection for this function

Reply:
Updated in this [commit](https://github.com/DistributedCollective/sovryn-lb-dex/pull/3/commits/364df5e45ac49cb94a7306ccc9835a9caf9b5cec)